pkg/vulnerabilityreport/plugin.go

```diff
@@ -23,7 +25,7 @@ type Plugin interface {
 	// The second argument maps container names to Docker registry credentials,
 	// which can be passed to the scanner as environment variables with values
 	// set from returned secrets.
-	GetScanJobSpec(ctx starboard.PluginContext, spec corev1.PodSpec, credentials map[string]docker.Auth) (
+	GetScanJobSpec(ctx starboard.PluginContext, obj client.Object, credentials map[string]docker.Auth) (
```

Comment:
To indicate that we expect K8s workloads, i.e. object that do have PodSpec template:
```suggestion
	GetScanJobSpec(ctx starboard.PluginContext, workload client.Object, credentials map[string]docker.Auth) (
```